cheroot/makefile.py

```diff
@@ -32,8 +34,44 @@ def _flush_unlocked(self):
                 n = self.raw.write(bytes(self._write_buf))
             except io.BlockingIOError as e:
                 n = e.characters_written
+            except OSError as sock_err:
+                error_code = sock_err.errno
+                if error_code in _errors.acceptable_sock_shutdown_error_codes:
+                    # The socket is gone, so just ignore this error.
+                    return
+                raise
+            else:
+                # The 'try' block completed without an exception
+                if n == 0:
+                    # This could happen with non-blocking write
+                    # when nothing was written
+                    break
+
             del self._write_buf[:n]
 
+    def close(self):
+        """
+        Close the stream and its underlying file object.
+
+        This method is designed to be idempotent (it can be called multiple
+        times without side effects). It gracefully handles a race condition
+        where the underlying socket may have already been closed by the remote
+        client or another thread.
+
+        A SysCallError or OSError with ``EBADF`` or ``ENOTCONN`` is caught
+        and ignored, as these indicate a normal, expected connection teardown.
+        Other exceptions are re-raised.
+        """
+        if self.closed:  # pylint: disable=W0125
```

Comment:
Use human-readable codes for pylint. A letter and a bunch of numbers don't really give away any context regarding when this can be dropped or why it's here is the first place.